core/templates/pages/topic-editor-page/preview-tab/topic-preview-tab.component.spec.ts

```diff
@@ -16,76 +16,94 @@
  * @fileoverview Unit tests for topic preview tab.
  */
 
-import { HttpClientTestingModule } from '@angular/common/http/testing';
-import { TestBed } from '@angular/core/testing';
+import { NO_ERRORS_SCHEMA } from '@angular/core';
+import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
+import { MaterialModule } from 'components/material.module';
+import { StorySummary } from 'domain/story/story-summary.model';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+import { TopicEditorStateService } from '../services/topic-editor-state.service';
+import { TopicPreviewTabComponent } from './topic-preview-tab.component';
 
-import { EditableStoryBackendApiService } from 'domain/story/editable-story-backend-api.service';
-import { StorySummary} from 'domain/story/story-summary.model';
-import { importAllAngularServices } from 'tests/unit-test-utils';
+describe('Topic Preview Tab Component', () => {
+  let fixture: ComponentFixture<TopicPreviewTabComponent>;
+  let componentInstance: TopicPreviewTabComponent;
+  let testName = 'test_name';
+  let mockUrl = 'mock_url';
+  let storySummaries = [new StorySummary(
+    null, null, [], null, null, null, null, [], null, [])];
 
-describe('Topic preview tab', function() {
-  var ctrl = null;
-  var $rootScope = null;
-  var $scope = null;
-  var TopicEditorStateService = null;
+  class MockTopicEditorStateService {
+    getTopic() {
+      return {
+        getName(): string {
+          return testName;
+        },
+        getSubtopics() {
+          return [];
+        },
+      };
+    }
 
-  importAllAngularServices();
+    getCanonicalStorySummaries() {
+      return storySummaries;
+    }
+  }
 
-  beforeEach(angular.mock.module('oppia'));
-  beforeEach(() => {
+  class MockUrlInterpolationService {
+    getStaticImageUrl(imagePath: string): string {
+      return mockUrl;
+    }
+  }
```

Comment:
[Optional] Can you `spyOn` in beforeEach  instead?

Reply:
Using a `spyOn` will add the necessity to explicitly typecast the mock object as TopicEditorStateService (similarly, for other services).